corehq/apps/app_manager/models.py

```diff
@@ -4487,7 +4538,10 @@ def make_build(self, comment=None, user_id=None, previous_version=None):
             # but check explicitly so as not to change the _id if it exists
             copy._id = copy.get_db().server.next_uuid()
 
-        copy.set_form_versions(previous_version)
+        force_new_forms = False
+        if previous_version and self.build_profiles != previous_version.build_profiles:
+            force_new_forms = True
```

Comment:
We need to force the phone to re fetch the form, since the languages may have changed even if the form has not, otherwise the phone will not get the new languages. This mainly happens matters when a new language is added to a profile or a profile is deleted and the app reverts to the default build. In both those cases the app wouldnt get the other translations of the form because the version numbers would be the same.